driver-core/src/main/com/mongodb/connection/ResponseBuffers.java

```diff
@@ -41,6 +43,12 @@ public ReplyHeader getReplyHeader() {
         return replyHeader;
     }
 
+    <T extends BsonDocument> T getResponseDocument(final int messageId, final Decoder<T> decoder) {
```

Comment:
Moved this here from a static method in InternalStreamConnection so it can also be used by LoggingCommandEventSender